dbms/src/Storages/StorageGenerateRandom.cpp

```diff
@@ -0,0 +1,396 @@
+#include <Storages/IStorage.h>
+#include <Storages/ColumnsDescription.h>
+#include <Storages/StorageGenerateRandom.h>
+#include <Storages/StorageFactory.h>
+#include <Processors/Sources/SourceFromSingleChunk.h>
+#include <Processors/Pipe.h>
+#include <Parsers/ASTLiteral.h>
+
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeDecimalBase.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnNullable.h>
+#include <Columns/ColumnTuple.h>
+
+#include <Common/SipHash.h>
+#include <Common/randomSeed.h>
+#include <common/unaligned.h>
+
+#include <Functions/FunctionFactory.h>
+
+#include <pcg_random.hpp>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+
+namespace
+{
+
+void fillBufferWithRandomData(char * __restrict data, size_t size, pcg64_fast & rng)
+{
+    char * __restrict end = data + size;
+    while (data < end)
+    {
+        /// The loop can be further optimized.
+        UInt64 number = rng();
+        unalignedStore<UInt64>(data, number);
+        data += sizeof(UInt64); /// We assume that data has 15-byte padding (see PaddedPODArray)
+    }
+}
+
+
+ColumnPtr fillColumnWithRandomData(
+    const DataTypePtr type, UInt64 limit, UInt64 max_array_length, UInt64 max_string_length, pcg64_fast & rng, const Context & context)
+{
+    TypeIndex idx = type->getTypeId();
+
+    switch (idx)
+    {
+        case TypeIndex::String:
+        {
+            auto size_column = ColumnUInt32::create();
+            auto & sizes = size_column->getData();
+
+            sizes.resize(limit);
+            for (UInt64 i = 0; i < limit; ++i)
+                sizes[i] = static_cast<UInt32>(rng()) % max_string_length;  /// Slow
+
+            ColumnWithTypeAndName argument{std::move(size_column), std::make_shared<DataTypeUInt32>(), "size"};
+
+            Block block
+            {
+                argument,
+                {nullptr, type, "result"}
+            };
+
+            FunctionFactory::instance().get("randomPrintableASCII", context)->build({argument})->execute(block, {0}, 1, limit);
```

Comment:
We can't use randomPrintableASCII as it has no way do have deterministic result. There is no way to set random_seed.

Reply:
Ok.